common/configuration/1password_provider.go

```diff
@@ -36,16 +51,46 @@ func (o OnePasswordProvider) Load(ctx context.Context, ref Ref, key *url.URL) ([
 		return nil, fmt.Errorf("1Password secret reference must be a base64 encoded string: %w", err)
 	}
 
-	output, err := exec.Capture(ctx, ".", "op", "read", "-n", string(decoded))
+	parsedRef, err := decodeSecretRef(string(decoded))
+	if err != nil {
+		return nil, fmt.Errorf("1Password secret reference invalid: %w", err)
+	}
+
+	args := []string{"--format", "json", "item", "get", "--vault", parsedRef.Vault, parsedRef.Item}
+	v, fieldSpecified := parsedRef.Field.Get()
+	if fieldSpecified {
+		args = append(args, "--fields", v)
+	}
+	output, err := exec.Capture(ctx, ".", "op", args...)
 	if err != nil {
-		lines := bytes.Split(output, []byte("\n"))
-		logger := log.FromContext(ctx)
-		for _, line := range lines {
-			logger.Warnf("%s", line)
+		return nil, fmt.Errorf("run `op` with args %v: %w", args, err)
+	}
+
+	if fieldSpecified {
+		v, err := decodeSingle(output)
+		if err != nil {
+			return nil, err
 		}
-		return nil, fmt.Errorf("error running 1password CLI tool \"op\": %w", err)
+
+		return json.Marshal(v.Value)
+	}
+
+	full, err := decodeFull(output)
+	if err != nil {
+		return nil, err
 	}
-	return json.Marshal(string(output))
+
+	// Filter out anything without a value
+	filtered := slices.Filter(full, func(e entry) bool {
+		return e.Value != ""
+	})
+	// Map to id: value
+	var mapped = make(map[string]string)
```

Comment:
If you know the number of final entries, it's typical to set the capacity with eg. `make(map[string]string, len(filtered))`